### ISSUE TITLE
Differences in the results of the replication experiment

### ISSUE THREAD

**Issue by cq-dong** [zwyang6/ExCEL#5]
Hello, thank you for your excellent work in open source.
I would like to ask about the problem of reproducing experimental results. Is the test result reported in your paper the model of the last iteration of training or the best performing model among all iterations?
I have seen differences in the final results in both experiments. The highest miou in train.log on voc is 75.01, and the test results of the last iteration model are 0.766 and 0.767. Can you provide me with suggestions for reproduction based on the training log?

[train.log](https://github.com/user-attachments/files/21134215/train.log)
[train2.log](https://github.com/user-attachments/files/21134216/train2.log)

[results.log](https://github.com/user-attachments/files/21134224/results2.log)
[results2.log](https://github.com/user-attachments/files/21134225/results.log)

og)

**Comment by zwyang6** [comment]
hi, thanks for your attention!

In my experiments, the best reuslt, i.e., the seg performance im log file, are mostly originated from the checkpoint at last iteration. I notice the best seg result of your train.log file is 75.3, which is close to the my result 75.6. You can select the checkpoint at 28000 iter for the final evaluation.

**Comment by zwyang6** [comment]
But I notice your performance of pseudo labels is lower from my results, about by 1% miou. It could be the difference of our implementing enviroment or something i missed when i sorted my code. I will carefully check my code recently. 

If you find it helpful, A star is appreciated^-^

**Comment by diaodiao317** [comment]
I did not receive the earlier GitHub notifications and unfortunately missed your previous reply to my question—my apologies for not responding in time. This user raised a question similar to mine, so I hope you don’t mind me replying here.

Thanks to the efficiency of your code, I was able to reproduce ExCEL five times. However, the validation performance remained relatively low, consistently around 77.1–77.5 after applying CRF. Due to the submission limit on the official test set, I only submitted once using your method, and the official result was 78.1.

I suspect the discrepancy might be related to the PyTorch version. However, modifying the environment on our computing cluster is quite complicated, so I haven’t tested with other versions. In our paper, we would like to report the reproduction results under our current environment, especially since we also attempted to plug-and-play our method into yours.

Thank you again for your open-source contribution, and I hope you understand our situation.

<img width="441" height="694" alt="Image" src="https://github.com/user-attachments/assets/77b35b62-9781-451a-b7ce-7a6efe016ec4" />

**Comment by zwyang6** [comment]
> I did not receive the earlier GitHub notifications and unfortunately missed your previous reply to my question—my apologies for not responding in time. This user raised a question similar to mine, so I hope you don’t mind me replying here.
> 
> Thanks to the efficiency of your code, I was able to reproduce ExCEL five times. However, the validation performance remained relatively low, consistently around 77.1–77.5 after applying CRF. Due to the submission limit on the official test set, I only submitted once using your method, and the official result was 78.1.
> 
> I suspect the discrepancy might be related to the PyTorch version. However, modifying the environment on our computing cluster is quite complicated, so I haven’t tested with other versions. In our paper, we would like to report the reproduction results under our current environment, especially since we also attempted to plug-and-play our method into yours.
> 
> Thank you again for your open-source contribution, and I hope you understand our situation.
> 
> <img alt="Image" width="441" height="694" src="https://private-user-images.githubusercontent.com/56783070/464635020-77b35b62-9781-451a-b7ce-7a6efe016ec4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTIxNDA4NzgsIm5iZiI6MTc1MjE0MDU3OCwicGF0aCI6Ii81Njc4MzA3MC80NjQ2MzUwMjAtNzdiMzViNjItOTc4MS00NTFhLWI3Y2UtN2E2ZWZlMDE2ZWM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzEwVDA5NDI1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI0MGRiYzliMzY4NmIzZmQwODFlMmE3MzhiMzc1NWYxYTRkOWE2NzI3NDkxMjI2M2FmM2M1NDNkNDg3ZjM3YjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xg5KVw046ux6WUBOQLTIYFpwn9MScwwtKlumILdbcLQ">


hi, thanks for your interest in our paper. We feel honored that our method can be compared as a basline to your upcoming work. 
 
I think it is okay to report the reproduced result as long as there is no abvious difference between these results. I think 78.1 is very close to our reported performance and I believe reviewers will not judge your work merely because the discrepancy probaly caused by the different Pytorch or CUDA version.  

Looking forward to your great upcoming work! ^^

**Comment by caoer01** [comment]
> I did not receive the earlier GitHub notifications and unfortunately missed your previous reply to my question—my apologies for not responding in time. This user raised a question similar to mine, so I hope you don’t mind me replying here.
> 
> Thanks to the efficiency of your code, I was able to reproduce ExCEL five times. However, the validation performance remained relatively low, consistently around 77.1–77.5 after applying CRF. Due to the submission limit on the official test set, I only submitted once using your method, and the official result was 78.1.
> 
> I suspect the discrepancy might be related to the PyTorch version. However, modifying the environment on our computing cluster is quite complicated, so I haven’t tested with other versions. In our paper, we would like to report the reproduction results under our current environment, especially since we also attempted to plug-and-play our method into yours.
> 
> Thank you again for your open-source contribution, and I hope you understand our situation.
> 
> <img alt="Image" width="441" height="694" src="https://private-user-images.githubusercontent.com/56783070/464635020-77b35b62-9781-451a-b7ce-7a6efe016ec4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI4NTM4NzgsIm5iZiI6MTc2Mjg1MzU3OCwicGF0aCI6Ii81Njc4MzA3MC80NjQ2MzUwMjAtNzdiMzViNjItOTc4MS00NTFhLWI3Y2UtN2E2ZWZlMDE2ZWM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTExVDA5MzI1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5MGJlMzlhNDY2ODYzYjBhM2YwN2Q3ZjZhMmQ2Y2UzOWExZTc3ODhjMDc3Y2Y3YTkyMTA3Y2MxMDhhNDcyZmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5yrUGmRVxTqUgUtcGAk72Vw9PXLsEOivMo8D9uiuEr0">

我也一样，应用crf的结果，验证集77.2，是论文中报道没有crf的结果，而不用crf，验证集结果76.3。我的环境包pytorch cuda版本等是和作者的requirement里一样，但是我是4090 24g跑的。
